domains/certificates/Query.API/API.AppTests/Extensions/HttpClientExtensions.cs

```diff
@@ -21,12 +21,14 @@ public static async Task<T> RepeatedlyGetUntil<T>(this HttpClient client, string
 
         do
         {
-            var response = await client.GetAsync(requestUri);
-            if (response.StatusCode == HttpStatusCode.OK)
+            using (var response = await client.GetAsync(requestUri))
             {
-                var content = await response.Content.ReadFromJsonAsync<T>();
-                if (content != null && condition(content))
-                    return content;
+                if (response.StatusCode == HttpStatusCode.OK)
+                {
+                    var content = await response.Content.ReadFromJsonAsync<T>();
+                    if (content != null && condition(content))
```

Comment:
Could this condition be separated in a method describing what it does?

Reply:
`condition` is a Func delegate and this is could be named to something else. Do you have a suggestion?

Reply:
No sorry as i don't know what it actually checks for.

Reply:
The extention method is called `RepeatedlyGetUntil` and it will repeatedly do a GET request until the condition given in the parameter `condition` is met/true.  Please note that the condition is supplied outside of the method, so what it actually does is defined elsewhere.